skins/Belchertown/js/belchertown.js.tmpl

```diff
@@ -386,6 +397,34 @@ function beaufort_cat(beaufort) {
     }
 }
 
+#if $Extras.has_key("airfield_crosswind_enabled") and $Extras.airfield_crosswind_enabled == '1'
+const active_runway = parseFloat($Extras.airfield_crosswind_active_runway).toFixed(0);
+
+// Find Opposite Runway
+var opposite_runway = (((active_runway * 10) + 180) % 360) / 10;
+
+// Positionning Active Runway and Opposite at the right place
+var css_north = opposite_runway;
+var css_south = active_runway;
+
+function calcXwind(windDir, windspeed, runway) {
```

Comment:
I don't know if your function was always wrong but for wind 360/10 on RWY 36, it gives the wrong result.

Here is my implementation, read the comments for more:

`function calcXwind(winddir, windspeed, runway) {
    // Use float values as that will provide better precision with the triginometric functions below.
    winddir = parseFloat(winddir);
    windspeed = parseFloat(windspeed);
    
    // The angle alpha is the difference between the wind and the active runway, in degrees. Use (Math.PI / 180) to convert to radians more precise.
    const alpha = (winddir - runway * 10) * (Math.PI / 180);

    // If the headwind becomes negative, then you have not fed the active runway into the function.
    const headwind = Math.round(windspeed * Math.cos(alpha));

    // Positive crosswind is from the right, negative is from the left.
    const crosswind = Math.round(windspeed * Math.sin(alpha));
    const direction = crosswind == 0 ? 'None' : crosswind > 0 ? 'Right' : 'Left';

    // Show the values
    jQuery(".headwind").html(headwind);
    jQuery(".crosswind").html(Math.abs(crosswind));
    jQuery(".crosswind_lrxw").html(direction);
}`

Reply:
Thank you for the interest in this modification,
I had not seen this error before, and will try your code snippet / optimization asap.
On other part I'm working on a better css / html integration.

Reply:
I'm comparing results between your function and mine and have same final results so I can't see where the error. Any way your code is cleanest, faster, and more precise.

Reply:
Changes merged https://github.com/poblabs/weewx-belchertown/pull/657/commits/597485f2dd1c82cf7c94dd435309e091b34e1355